### PR TITLE
Add event edit command

### PR DESCRIPTION
**Pull request by tiongMax**
Implement eedit feature. 

Test cases are done.

Pls review and merge if it is all good.

### CODE REVIEW

**Review by jordankanghm — APPROVED** [review]
looks amazing to me bro

File: src/main/java/seedu/address/logic/commands/eventcommands/EventEditCommand.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+
+/**
+ * Edits the details of an existing event in the address book.
+ */
+public class EventEditCommand extends Command {
+
+    public static final String COMMAND_WORD = "eedit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the edit identified "
```
Comment (zacwong2151) [inline src/main/java/seedu/address/logic/commands/eventcommands/EventEditCommand.java:44]: think you meant: Edits the details of the `event` identified

File: src/main/java/seedu/address/logic/commands/eventcommands/EventEditCommand.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+            + "[" + PREFIX_ROLE + "ROLES] "
+            + "[" + PREFIX_START_DATETIME + "START DATE] "
+            + "[" + PREFIX_LOCATION + "LOCATION] ...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_ROLE + "Director "
+            + PREFIX_LOCATION + "Kent Ridge";
+
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited Event: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in the address book.";
```
Comment (zacwong2151) [inline src/main/java/seedu/address/logic/commands/eventcommands/EventEditCommand.java:58]: think you meant: This event already exists in the `event list`

File: src/main/java/seedu/address/logic/commands/eventcommands/EventEditCommand.java [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+import seedu.address.model.event.Event;
+import seedu.address.model.event.EventName;
+import seedu.address.model.event.Location;
+import seedu.address.model.event.Material;
+import seedu.address.model.event.Role;
+import seedu.address.model.volunteer.Name;
+
+
+/**
+ * Edits the details of an existing event in the address book.
```
Comment (zacwong2151) [inline src/main/java/seedu/address/logic/commands/eventcommands/EventEditCommand.java:38]: `event list`, not `address book`

**Review by zacwong2151 — APPROVED** [review]
So many lines of code sickk 
Just have some small typos to fix then can go ahead and merge

**Review by brein62 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/eventcommandparsers/EventEditCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    private Optional<Set<Material>> parseMaterialsForEdit(Collection<String> materials) throws ParseException {
+        assert materials != null;
+
+        if (materials.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> materialSet = materials.size() == 1 && materials.contains("") ? Collections.emptySet()
+                                                                                            : materials;
+        return Optional.of(ParserUtil.parseMaterials(materialSet));
+    }
```
Comment (brein62) [inline src/main/java/seedu/address/logic/parser/eventcommandparsers/EventEditCommandParser.java:135]: not sure but for your reference if some test cases fail due to the change in Material

Comment (brein62) [reply src/main/java/seedu/address/logic/parser/eventcommandparsers/EventEditCommandParser.java:135]: unless you wanna make more changes

Comment (tiongMax) [reply src/main/java/seedu/address/logic/parser/eventcommandparsers/EventEditCommandParser.java:135]: nah it is not ur fault, i lost my cool.

**Pull request merged**